### PR TITLE
(maint), (PE-35980) Modernization updates for Puppet 8 and newer libgit2 proxy usage

### PR DESCRIPTION
**Pull request by justinstoller**
Please add all notable changes to the "Unreleased" section of the CHANGELOG in the format:
```
- (JIRA ticket) Summary of changes. [Issue or PR #](link to issue or PR)
```

### CODE REVIEW

**Comment by justinstoller** [comment]
I'm out for a long 4th of July vacation, but this is approximately what is needed to unblock r10k CI using Puppet 8 and modern dependencies in preparation for a new release.

File: integration/tests/git_source/git_source_git.rb
Comment (bastelfreak) [inline integration/tests/git_source/git_source_git.rb]: wow that's hacky :D

**Comment by justinstoller** [comment]
I think everything is good to go now, _execpt_ the last commit which is waiting on this PR and a release to our internal forge https://github.com/puppetlabs/puppetlabs-xinetd/pull/125

**Comment by justinstoller** [comment]
Okay, now that a new version of xinetd is on the forge I've moved us to use that. It is successful running here:
```
      Test Suite: tests @ 2023-07-18 13:41:07 -0700

      - Host Configuration Summary -


              - Test Case Summary for suite 'tests' -
       Total Suite Time: 1985.83 seconds
      Average Test Time: 30.09 seconds
              Attempted: 66
                 Passed: 62
                 Failed: 0
                Errored: 0
                Skipped: 4
                Pending: 0
                  Total: 66

      - Specific Test Case Status -
        
Failed Tests Cases:
Errored Tests Cases:
Skipped Tests Cases:
  Test Case tests/basic_functionality/rugged_git_provider_with_ssh.rb 
  Test Case tests/git_source/git_source_ssh.rb 
  Test Case tests/i18n/deploy_module_with_unicode_in_file_name.rb 
  Test Case tests/user_scenario/basic_workflow/multi_env_hiera.rb 
Pending Tests Cases:

No tests to run for suite 'post_suite'
No tests to run for suite 'pre_cleanup'
Cleanup: cleaning up after successful run
```
With that I'm going to remove the patching of the installed r10k and rebase.

**Comment by justinstoller** [comment]
Ooooooooookay. I think this has everything we finally want/need for our updates, I am running the integration tests locally right now against rugged and shellgit backends and will report back. :crossed_fingers: we can get this merged today!!

**Comment by justinstoller** [comment]
For GIT_PROVIDER=rugged I get:
```

              - Test Case Summary for suite 'tests' -
       Total Suite Time: 1914.43 seconds
      Average Test Time: 29.01 seconds
              Attempted: 66
                 Passed: 61
                 Failed: 2
                Errored: 0
                Skipped: 3
                Pending: 0
                  Total: 66

      - Specific Test Case Status -
        
Failed Tests Cases:
  Test Case tests/Puppetfile/HTTP_PROXY_affects_git_source.rb reported: #<Minitest::Assertion: expected error code was not observed>
    Test line: tests/Puppetfile/HTTP_PROXY_affects_git_source.rb:73:in `block (2 levels) in run_test'
  Test Case tests/git_source/HTTP_proxy_and_git_source.rb reported: #<Minitest::Assertion: The expected error message was not observed.
Expected /failed to resolve address for foooooooo\.unresolvable/ to match "ERROR\t -> Unable to determine current branches for Git source 'control' (/etc/puppetlabs/code/environments)\nOriginal exception:\nfailed to resolve address for something.else: Name or service not known at /var/cache/r10k/https---something.else-repo\n".>
    Test line: tests/git_source/HTTP_proxy_and_git_source.rb:68:in `block (2 levels) in run_test'
Errored Tests Cases:
Skipped Tests Cases:
  Test Case tests/basic_functionality/rugged_git_provider_with_ssh.rb 
  Test Case tests/git_source/git_source_ssh.rb 
  Test Case tests/user_scenario/basic_workflow/multi_env_hiera.rb 
```
which are expected proxy failures until the proxy updates in this PR are merged. Confirming a clean run with shellgit, right now.

**Comment by justinstoller** [comment]
Shellgit had a single failure:
```
      Test Suite: tests @ 2023-07-18 16:20:46 -0700

      - Host Configuration Summary -


              - Test Case Summary for suite 'tests' -
       Total Suite Time: 1977.75 seconds
      Average Test Time: 29.97 seconds
              Attempted: 66
                 Passed: 60
                 Failed: 0
                Errored: 1
                Skipped: 5
                Pending: 0
                  Total: 66

      - Specific Test Case Status -
        
Failed Tests Cases:
Errored Tests Cases:
  Test Case tests/git_source/negative/neg_git_unauthorized_ssh.rb reported: #<Beaker::Host::CommandFailure: Host 'clever-pitch.delivery.puppetlabs.net' exited with 0 running:
 /opt/puppetlabs/puppet/bin/r10k deploy environment -v
Last 10 lines of output were:
	INFO	-> Deploying environment /etc/puppetlabs/code/environments/production
	INFO	-> Environment production is now at 49f60cfd51e066a76ce3cf2d7eda2e019dedc562>
    Test line: tests/git_source/negative/neg_git_unauthorized_ssh.rb:71:in `block in run_test'
Skipped Tests Cases:
  Test Case tests/basic_functionality/rugged_git_provider_with_ssh.rb 
  Test Case tests/basic_functionality/rugged_git_provider_without_ssh.rb 
  Test Case tests/git_source/git_source_ssh.rb 
  Test Case tests/user_scenario/basic_workflow/multi_env_hiera.rb 
  Test Case tests/user_scenario/basic_workflow/negative/neg_bad_git_module.rb 
```
Re-running just this test to see if it still fails

**Comment by justinstoller** [comment]
With the last update GIT_PROVIDER=shellgit now passes tests/git_source/negative/neg_git_unauthorized_ssh
```
tests/git_source/negative/neg_git_unauthorized_ssh.rb passed in 5.87 seconds
      Test Suite: tests @ 2023-07-19 10:57:59 -0700

      - Host Configuration Summary -


              - Test Case Summary for suite 'tests' -
       Total Suite Time: 5.87 seconds
      Average Test Time: 5.87 seconds
              Attempted: 1
                 Passed: 1
                 Failed: 0
                Errored: 0
                Skipped: 0
                Pending: 0
                  Total: 1

      - Specific Test Case Status -
        
Failed Tests Cases:
Errored Tests Cases:
Skipped Tests Cases:
Pending Tests Cases:

```

**Review by bastelfreak — APPROVED** [review]

**Review by nmburgan — APPROVED** [review]
woohoo

**Pull request merged**
